python/neuroglancer/dvid_credentials.py

```diff
@@ -0,0 +1,59 @@
+# @license
+# Copyright 2021 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Module implements function for authentication of layers based on DVID.
+    Here tokens are fetched from local locations like env vars etc."""
+
+from __future__ import absolute_import
+
+import logging
+import os
+
+from . import credentials_provider
+from .futures import run_on_new_thread
+
+
+class TokenbasedDefaultCredentialsProvider(credentials_provider.CredentialsProvider):
+    def __init__(self):
+        super(TokenbasedDefaultCredentialsProvider, self).__init__()
+
+        # Make sure logging is initialized.
+        # Does nothing if logging has already been initialized.
+        logging.basicConfig()
+
+        self._credentials = {}
+
+    def get_new(self):
+        def func():
+            self._credentials = {}
+            try:
+                TOKEN = os.environ['DVID_APPLICATION_CREDENTIALS']
```

Comment:
Actually, now that I think about this a bit more, I am worried that this makes it very easy to accidentally leak your credentials to the wrong server.  If you add multiple DVID layers for different servers, you may send the authentication token intended for one server to a different server.

Is DVID_APPLICATION_CREDENTIALS a standard environment variable or did you just come up with that?

Perhaps instead you could use something like:

DVID_CREDENTIALS="server.com:TOKEN"

or

DVID_CREDENTIALS="server1.com:TOKEN1 server2.com:TOKEN2"

Reply:
Actually, for consistency with HTTP origins it would be better to allow a port number for the server as well, which means colon doesn't work as a separator.  Instead, you could use: DVID_CREDENTIALS="server1.com=TOKEN1 server2.com:8000=TOKEN2"